backend/file_management/file_management_helper.py

```diff
@@ -165,19 +181,26 @@ def fetch_file_contents(file_system: UnstractFileSystem, file_path: str) -> Any:
         except ApiRequestError as exception:
             logger.error(f"ApiRequestError from {file_info} {exception}")
             raise ConnectorApiRequestError
+
+        data = ""
+        # Check if the file type is in the allowed list
+        if file_content_type not in allowed_content_types:
+            raise InvalidFileType(f"File type '{file_content_type}' is not allowed.")
+
+        # Handle allowed file types
         if file_content_type == "application/pdf":
-            # Read contents of PDF file into a string
             with fs.open(file_path, "rb") as file:
-                encoded_string = base64.b64encode(file.read())
-                return encoded_string
+                data = base64.b64encode(file.read())
 
         elif file_content_type == "text/plain":
             with fs.open(file_path, "r") as file:
                 logger.info(f"Reading text file: {file_path}")
-                text_content = file.read()
-                return text_content
+                data = file.read()
+
         else:
-            raise InvalidFileType
+            raise InvalidFileType(f"File type '{file_content_type}' is not handled.")
```

Comment:
@jagadeeswaran-zipstack what's the difference between file types we haven't allowed and file types which we haven't handled? Do you think we need such separate errors to be thrown?

Reply:
Like we discussed, Instead of throwing error added a warning log there.